pkg/apis/gke.cattle.io/v1/types.go

```diff
@@ -33,105 +33,300 @@ type GKEClusterConfig struct {
-	NodeIpv4CidrBlock          string `json:"nodeIpv4CidrBlock,omitempty"`
-	ServicesIpv4CidrBlock      string `json:"servicesIpv4CidrBlock,omitempty"`
+	// +optional
+	// ClusterIpv4CidrBlock is the IPv4 CIDR block for the cluster.
+	ClusterIpv4CidrBlock string `json:"clusterIpv4CidrBlock,omitempty"`
+
+	// +optional
+	// ClusterSecondaryRangeName is the name of the secondary range for cluster pods.
+	ClusterSecondaryRangeName string `json:"clusterSecondaryRangeName,omitempty"`
+
+	// +optional
+	// CreateSubnetwork indicates whether to create a subnetwork for the cluster.
+	CreateSubnetwork bool `json:"createSubnetwork,omitempty"`
+
+	// +optional
+	// NodeIpv4CidrBlock is the IPv4 CIDR block for nodes.
+	NodeIpv4CidrBlock string `json:"nodeIpv4CidrBlock,omitempty"`
+
+	// +optional
+	// ServicesIpv4CidrBlock is the IPv4 CIDR block for services.
+	ServicesIpv4CidrBlock string `json:"servicesIpv4CidrBlock,omitempty"`
+
+	// +optional
+	// ServicesSecondaryRangeName is the name of the secondary range for services.
 	ServicesSecondaryRangeName string `json:"servicesSecondaryRangeName,omitempty"`
-	SubnetworkName             string `json:"subnetworkName,omitempty"`
-	UseIPAliases               bool   `json:"useIpAliases,omitempty"`
+
+	// +optional
+	// SubnetworkName is the name of the subnetwork to use, if specified.
+	SubnetworkName string `json:"subnetworkName,omitempty"`
+
+	// +optional
+	// UseIPAliases indicates whether to use IP aliases.
+	UseIPAliases bool `json:"useIpAliases,omitempty"`
 }
 
 type GKEPrivateClusterConfig struct {
-	EnablePrivateEndpoint bool   `json:"enablePrivateEndpoint,omitempty"`
-	EnablePrivateNodes    bool   `json:"enablePrivateNodes,omitempty"`
-	MasterIpv4CidrBlock   string `json:"masterIpv4CidrBlock,omitempty"`
+	// +optional
+	// EnablePrivateEndpoint enables the private endpoint for the cluster.
+	EnablePrivateEndpoint bool `json:"enablePrivateEndpoint,omitempty"`
+
+	// +optional
+	// EnablePrivateNodes enables private nodes for the cluster.
+	EnablePrivateNodes bool `json:"enablePrivateNodes,omitempty"`
+
+	// +kubebuilder:validation:Required
+	// MasterIpv4CidrBlock is the IPv4 CIDR block for the master.
+	MasterIpv4CidrBlock string `json:"masterIpv4CidrBlock,omitempty"`
 }
 
 type GKEClusterConfigStatus struct {
-	Phase          string `json:"phase"`
-	FailureMessage string `json:"failureMessage"`
+	// +optional
+	// Phase represents the current phase of the cluster.
+	Phase string `json:"phase"`
+
+	// +optional
+	// FailureMessage contains an optional failure message for the cluster.
+	FailureMessage string `json:"failureMessage,omitempty"`
 }
 
 type GKEClusterAddons struct {
-	HTTPLoadBalancing        bool `json:"httpLoadBalancing,omitempty"`
+	// +optional
+	// HTTPLoadBalancing indicates whether HTTP load balancing is enabled.
+	HTTPLoadBalancing bool `json:"httpLoadBalancing,omitempty"`
+
+	// +optional
+	// HorizontalPodAutoscaling indicates whether horizontal pod autoscaling is enabled.
```

Comment:
// +kubebuilder:default=true

Reply:
this and all below field where you are suggesting we should set `// +kubebuilder:default=true` seems to be `false` by default based on https://github.com/rancher/terraform-provider-rancher2/blob/master/docs/resources/cluster.md#gke_config_v2?